src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,47 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Duke {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        this.taskList = new TaskList(storage);
+    }
+
+    public void run() {
+        ui.showWelcomeMessage();
+        try {
+            Scanner sc = new Scanner(storage.load());
+            String input = sc.nextLine();
+            Parser parser = new Parser(taskList);
+            
+            while (sc.hasNextLine()) {
+                parser.execute(input);
```

Comment:
perhaps you can change the name of the method since a parser is parsing an input

```suggestion
                parser.parseInput(input);
```